### PR TITLE
Relay API changes for "by name" and accessing description

### PR DESCRIPTION
**Pull request by lhaskins**
This PR covers changes for show, update and delete when the relay name is supplied.
It also adds the relay description to the relay model.

Fixes https://github.com/operable/cog/issues/487

### CODE REVIEW

File: test/unit/cog_api/http/relays_test.exs [hunk 1]
```diff
@@ -55,6 +56,26 @@ defmodule CogApi.HTTP.RelaysTest do
         assert created_relay.enabled == true
       end
     end
+
+    context "when given a relay name" do
+      it "returns a relay" do
+        cassette "relay_show" do
```
Comment (drapergeek) [inline test/unit/cog_api/http/relays_test.exs]: Should this be a different casset name?

File: test/unit/cog_api/http/relays_test.exs [hunk 2]
```diff
@@ -89,13 +110,15 @@ defmodule CogApi.HTTP.RelaysTest do
...
 
-        updated = Client.relay_update(
-          new_relay.id,
-          %{name: "updated"},
-          endpoint
-        ) |> get_value
+        assert new_relay.description == nil
+        Client.relay_index(endpoint)
+
+        updated = Client.relay_update(new_relay.id,
```
Comment (drapergeek) [inline test/unit/cog_api/http/relays_test.exs]: What do you think about breaking this like:

```elixir
  updated = Client.relay_update(
    new_relay.id,
   %{description: "Hello"},
    endpoint
  )
```
```

File: test/unit/cog_api/http/relays_test.exs [hunk 3]
```diff
@@ -137,5 +182,18 @@ defmodule CogApi.HTTP.RelaysTest do
...
+          endpoint = valid_endpoint
+          relay = Client.relay_create(%{name: "delete me", token: "1234"}, endpoint)
+          |> get_value
+
+          assert :ok == Client.relay_delete(%{name: relay.name}, endpoint)
+        end
+      end
+    end
   end
+
```
Comment (drapergeek) [inline test/unit/cog_api/http/relays_test.exs]: I think this can be removed.

**Comment by lhaskins** [comment]
@drapergeek Thanks for the comments. I added a couple of new cassettes and changed the calling style as you suggested.

File: test/unit/cog_api/http/relays_test.exs [hunk 4]
```diff
@@ -55,6 +56,26 @@ defmodule CogApi.HTTP.RelaysTest do
...
+          assert created_relay.id == found_relay.id
+          assert created_relay.name == found_relay.name
+          assert found_relay.enabled == true
+          assert found_relay.description == "This is a test"
+          assert present found_relay.inserted_at
+        end
+      end
+    end
+
```
Comment (drapergeek) [inline test/unit/cog_api/http/relays_test.exs]: I don't think you need this extra line here.

File: test/unit/cog_api/http/relays_test.exs [hunk 5]
```diff
@@ -115,6 +137,27 @@ defmodule CogApi.HTTP.RelaysTest do
...
+            %{name: new_relay.name},
+            %{description: "Hello"},
+            endpoint
+          ) |> get_value
+
+          assert updated.description == "Hello"
+        end
+      end
+    end
+
```
Comment (drapergeek) [inline test/unit/cog_api/http/relays_test.exs]: I don't think you need this extra line here.

**Comment by drapergeek** [comment]
2 small comments then :shipit:

**Pull request merged**